rcl/src/rcl/node.c

```diff
@@ -52,9 +55,39 @@ typedef struct rcl_node_impl_t
   rmw_node_t * rmw_node_handle;
   uint64_t rcl_instance_id;
   rcl_guard_condition_t * graph_guard_condition;
+  const char * logger_name;
 } rcl_node_impl_t;
 
 
+const char * rcl_create_node_logger_name(
+  const char * node_name,
+  const char * node_namespace,
+  const rcl_allocator_t * allocator)
+{
+  char * node_name_with_ns = rcutils_format_string(*allocator, "%s/%s", node_namespace, node_name);
+  if (NULL == node_name_with_ns) {
+    return NULL;
+  }
+  // remove leading slashes
+  while (0 == rcutils_find(node_name_with_ns, '/')) {
```

Comment:
as discussed, I think this can be optimized wo/ the need for a while loop.

Reply:
yeah, given the [input assumptions](https://github.com/ros2/rcl/pull/212/commits/462e97ffcf9eeb3d9fbc06ca4d11b85f242227dd), this can be simplified: https://github.com/ros2/rcl/pull/212/commits/ac25438f9e8afd052e647a13a4e95a81ad02547c